worker/apiconfigwatcher/manifold.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package apiconfigwatcher
+
+import (
+	"sort"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/voyeur"
+	"launchpad.net/tomb"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/worker"
+	"github.com/juju/juju/worker/dependency"
+)
+
+var logger = loggo.GetLogger("juju.worker.apiconfigwatcher")
+
+type ManifoldConfig struct {
+	AgentName          string
+	AgentConfigChanged *voyeur.Value
+}
+
+// Manifold returns a dependency.Manifold which wraps the an agent's
+// voyeur.Value is set whenever it the agent config is
+// changed. Whenever the API server addresses in the config change the
+// manifold will bounce itself.
+//
+// The manifold is intended to be a dependency for the api-caller
```

Comment:
so, just to talk this through to myself: once the minion has connected to the new address and checked sanity, it then writes apiserver addresses and trusts the dependency engine to tear everything down and bring it back up again?

that works for me; but I'm wondering whether this is a bit overenthusiastic re bouncing in other circumstances? doesn't seem quite right for *everything* to have to reconnect when (say) one of 3 apiserver addresses changes.